### ISSUE TITLE
E0060 needs to be updated to new format

### ISSUE THREAD

**Issue by jonathandturner** [rust-lang/rust#35215]
From: src/test/compile-fail/E0060.rs

Error E0060 needs a span_label, updating it from:

```
error[E0060]: this function takes at least 1 parameter but 0 parameters were supplied
  --> src/test/compile-fail/E0060.rs:16:14
   |
16 |     unsafe { printf(); } //~ ERROR E0060
   |              ^^^^^^^^
   |
   = note: the following parameter type was expected: *const u8
```

To:

```
error[E0060]: this function takes at least 1 parameter but 0 parameters were supplied
  --> src/test/compile-fail/E0060.rs:16:14
   |
16 |     unsafe { printf(); } //~ ERROR E0060
   |              ^^^^^^^^ expected at least 1 parameter
   |
   = note: the following parameter type was expected: *const u8
```

Bonus: There may be a better way of displaying the information from the note.  Could it be part of the label?

**Comment by birryree** [comment]
I will take this issue.

**Comment by birryree** [comment]
@jonathandturner I got my changes in and working, but I had to set an environment variable to see the new formatting style? My command to make sure my changes actually did what they were supposed to look like this.

    RUST_NEW_ERROR_FORMAT=true ./build/x86_64-apple-darwin/stage1/bin/rustc src/test/compile-fail/E0060.rs

Is this expected? If so, I think it would help other new people tackling the issues in #35233 to mention that in your [very helpful blog post](http://www.jonathanturner.org/2016/08/helping-out-with-rust-errors.html) because I was definitely a little confused when my changes would pass the unit test but output still looked like the old format.
